schema/me/conversation/index.js

```diff
@@ -190,7 +190,28 @@ export const ConversationFields = {
   last_message_open: {
     type: GraphQLString,
     description: "Timestamp if the user opened the last message, null in all other cases",
-    resolve: conversation => null, // eslint-disable-line no-unused-vars
+    resolve: (conversation, options, request, { rootValue: { accessToken } }) => {
+      const radiationMessageId = get(conversation, "_embedded.last_message.radiation_message_id")
+      const impulseParams = {
+        conversation_id: conversation.id,
+        radiation_message_id: radiationMessageId,
+        "expand[]": "deliveries",
+      }
+      return gravity.with(accessToken, { method: "POST" })("me/token", {
+        client_application_id: IMPULSE_APPLICATION_ID,
+      }).then(data => {
+        return impulse.with(data.token, { method: "GET" })(
+          `message_details`,
+          impulseParams
+        ).then(({ message_details }) => {
+          if (message_details.length === 0) {
+            return null
+          }
+          const relevantDelivery = message_details[0].deliveries.find(d => d.email === conversation.from_email)
```

Comment:
Oh, hadn’t seen this discussion yet. I think the ‘fetch impulse token’ request should be made into a data loader that caches the token for the duration of the query resolving.